src/N3DataFactory.js

```diff
@@ -12,11 +12,12 @@ let _blankNodeCounter = 0;
 const escapedLiteral = /^"(.*".*)(?="[^"]*$)/;
 
 // ## DataFactory singleton
+// Note: The default data factory does not set the token field of terms.
 const DataFactory = {
-  namedNode,
-  blankNode,
-  variable,
-  literal,
+  namedNode: iri => namedNode(iri),
+  blankNode: name => blankNode(name),
+  variable: name => variable(name),
+  literal: (name, datatype) => literal(name, datatype),
```

Comment:
If I understand you correctly, you want to pass the default constructor functions to the DataFactory as they are. This would mean, however, that the default behavior of the DataFactory changes to always creating Terms with an initialized { token } context. Which in turn will break some tests and is probably not needed in standard situations.

I was hesitant to put a condition into the Parser to check wether to pass the token/context argument to the factory or not because this would have an impact on the performance. The solution to simply ignore the second argument in the DataFactory seemed like a good compromise and allows the API users to define their own behavior regarding the context variable of a node, such as offsetting conditions or adding additional data if needed.

So should I revert this to the default functions and have the context always initialized with a token?

Reply:
> the default behavior of the DataFactory changes to always creating Terms with an initialized { token } context.

That's fine; objects can always have optional fields.

> Which in turn will break some tests

Those tests might be too strict; we can loosen them.

> I was hesitant to put a condition into the Parser to check wether to pass the token/context argument to the factory

Indeed; let's definitely not.

> The solution to simply ignore the second argument in the DataFactory

Aah okay. This actually shows me that we need 2 DataFactory classes then. One that discards the context, and one that keeps it. Let's make both first-class citizens.